troubleshooting/ETW/Readme.md

```diff
@@ -127,43 +127,69 @@ This file will not exceed the configured max size and will be circularly overwri
 
 #### Configuration
 
-Configuration is controlled by a file named `ApplicationInsightsDiagnostics.json`.
+Configuration is controlled by a file named `ApplicationInsightsDiagnostics.json` or environment variables.
```

Comment:
What would happen if both are set? Or what should the customer expect to happen?

And what if one of them is of incorrect value/format?

Reply:
Current design is as follows:

Read config from environment variables, if exception occurs while parsing the variables, `TryGetConfiguration` returns false. If environment variables are successfully parsed, i.e. `TryGetConfigFromEnvrionmentVariable` returns true, `TryGetConfiguration` returns true without executing `TryGetConfigFromJsonFile`.

For the corner case that users set config via environment variables *and* JSON file:
1. If config were set *correctly both* via environment variables and the JSON file, SDK takes the configs set by environment variables.
2. If the config were set *incorrectly* via environment variables, SDK takes the configs set by JSON file.

I intentionally leave out the description regarding the above behavior in README because my assumption is that users initially will only try one path or the other regarding config setting. The situation of configs were set via both environment variables and JSON file most likely will only happen if the users use one way, but didn't see the diagnostics file being written.

I am thinking instead of having this corner case documented in README, throwing early in code when configs are not successfully set as expected might be a more direct way to communicate with the users.

How do you feel about it?

Reply:
The design is good. I meant to say it's better to have a description in README regarding which take precedence. It could be a short note like this in the [Application Insights guide](https://learn.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core?tabs=netcore6):

> An connection string specified in code wins over the environment variable APPLICATIONINSIGHTS_CONNECTION_STRING, which wins over other options.

Another example [here](https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config#connection-string).

It could be in the `#### Two ways to enable self-diagnostics` section or wherever you see fit.

Reply:
I've added a note regarding your suggestion. Thank you for the examples.